File: backend/internals/models/inventory.go
package models

import "gorm.io/gorm"

type Inventory struct {
	gorm.Model

	Name         string  `json:"name"`
	Brand        string  `json:"brand"`
	ProductModel string  `json:"model"`
	Description  string  `json:"description"`

	Price    float64 `json:"price"`
	Offer    int     `json:"offer"`
	Quantity int     `json:"quantity"`

	CategoryID uint     `json:"category_id"`
	Category   Category `json:"category"`
}

File: frontend/types/inventory.ts
export interface Category {
  ID: number;
  name: string;
}

export interface Inventory {
  ID: number;

  name: string;
  brand: string;
  model: string;
  description: string;

  price: number;
  offer: number;
  quantity: number;

  category_id: number;

  category: Category;
}

export interface InventoryResponse {
  success: boolean;

  data: Inventory[];

  pagination: {
    total: number;
    page: number;
    limit: number;
  };

  filters: {
    search: string;
    category_id: string;
    sort_feild: string;
    sort_order: string;
  };
}
